$NetBSD: patch-CMakeLists.txt,v 1.2 2024/02/10 03:10:53 riastradh Exp $

Remove overzealous compiler options and let pkgsrc define the
stack protection to use.

--- CMakeLists.txt.orig	2023-02-20 08:21:28.000000000 +0000
+++ CMakeLists.txt
@@ -90,7 +90,6 @@ if(NOT MSVC)
 endif()
 
 check_c_compiler_flag("-Wshorten-64-to-32" HAVE_SHORTEN_64_TO_32)
-check_c_compiler_flag("-Werror -fstack-protector-all" HAVE_STACK_PROTECTOR_ALL)
 
 check_include_files(cbor.h HAVE_CBOR_H)
 check_include_files(endian.h HAVE_ENDIAN_H)
@@ -295,7 +294,6 @@ else()
 
 	add_compile_options(-Wall)
 	add_compile_options(-Wextra)
-	add_compile_options(-Werror)
 	add_compile_options(-Wshadow)
 	add_compile_options(-Wcast-qual)
 	add_compile_options(-Wwrite-strings)
@@ -316,10 +314,6 @@ else()
 		add_compile_options(-Wshorten-64-to-32)
 	endif()
 
-	if(HAVE_STACK_PROTECTOR_ALL)
-		add_compile_options(-fstack-protector-all)
-	endif()
-
 	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -g2")
 	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer")
 	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -D_FORTIFY_SOURCE=2")
